### DISCUSSION TITLE
How to Avoid Failed Installs?

### DISCUSSION THREAD

**Discussion by Ashkaan** [svrooij/WinTuner#126] [Q&A] [answered]
Hi there,

In order for these apps to install winget must be updated from version 1.2 in Windows 11.  That's easily accomplished by adding the app '9NBLGGH4NNS1' to be required.

The problem is timing.  Most of my apps will fail to install because this app was not updated before it attempted to install.

How is everyone dealing with this?  Is there a way to ensure these winget apps have a dependency (the winget app isn't showing as a potential target dependency)?

I must be missing something huge or else everyone would have failed installs.

**Comment by svrooij** [comment]
This module can be used on an already finished computer to package apps to Intune.

Getting the metadata and the download location is done by reading the Winget manifest repository, it does not even use the Winget app in most cases.

It does however use Winget on the client machine to detect if an app is installed on a machine, in case the installer is no MSI and the product code is not provided.
This will result in Intune reporting faulty installed, while the apps might ben installed correctly. 

And if course of uses Winget on the client if you specify you want to package as a script (which just generates a ps1 with Winget install ...

Do you have a better solution to check of an app is installed correctly? Maybe you could update the detection rules afterwards, or maybe we even want a parameter for that

**Comment by Ashkaan** [comment]
Oh, I'm so sorry.  Here's what I meant:

- Your app allows us to deploy apps via winget in Intune.
- Your app successfully adds the app to Intune (I tested it).
- Any app that I add to Intune fails to install because the version of winget that ships with Windows 11 is defective (v1.2).  It cannot reach the repo.

The question is: **_How can we ensure your apps do not install until the update to winget occurs?_**

I assume every single person using your app has this issue and would like this answer, so I'm surprised that no one seems to have brought it up.

**Reply by svrooij** [reply, marked as answer]
This module and the packages itself don't use Winget until the detection step (for none MSI installers).

- Suggestion: you change the detection method of the app, to something that works for that app
- option 2: you figure out which Powershell command it would need to update Winget, and I'll add it to the generated detection script

> [!NOTE]
> If the detection fails (only place where it uses Winget), the app is still installed but reports a failed install

**Comment by Ashkaan** [comment]
I'm so sorry, but I don't understand the options.

The detection method (meaning, whether it's been installed or not) is something that your app creates for the package.  That's not the problem.  It fails to install.  It doesn't fail to detect the install.

Are you saying that we should bake the upgrading of winget into the detection script?  That seems messy.  I'm wondering how anyone else is getting this to work.  What did I miss?

**Reply by svrooij** [reply]
> The detection method (meaning, whether it's been installed or not) is something that your app creates for the package. That's not the problem. It fails to install. It doesn't fail to detect the install.

it does not use winget during install, only during packaging and detection. So the issue must lie in the detection. If it fails to detect the app after install, it will report a failed install.

> Are you saying that we should bake the upgrading of winget into the detection script? That seems messy. I'm wondering how anyone else is getting this to work. What did I miss?

im not saying that is a good idea, just looking for alternatives.
Or we could package the WinGet installation script and run that as a dependency.

**Comment by Ashkaan** [comment]
Sorry, what I'm trying to say is that there is no issue with detection.

The install is actually failing.  I mean, the software is not on the computer at all.  Of course it's not going to detect.

**Reply by svrooij** [reply]
And what Im trying to say is that is you used wintuner to package the installer there is no use of Winget during install.

What package is failing? Did you try to test the created package?
https://wintuner.app/docs/wintuner-powershell/Test-WtIntuneWin

That will try to install the created package in the windows sandbox.

**Comment by Ashkaan** [comment]
All packages created with your app are failing.  I believe winget must be functioning on the target computer for the package to install, right?

I just tested with the test app and the test passed.

**Reply by svrooij** [reply]
I'm locking this issue.

I replied 3 times that winget is not used during install, don't know how I can explain that even furter
